bittensor/utils/subprocess_utils.py

```diff
@@ -0,0 +1,364 @@
+
+        while not is_commit_reveal_subprocess_ready() and attempt_count < 5:
+            time.sleep(3)
+            print("Waiting for commit_reveal subprocess to be ready.")
+            attempt_count += 1
+
+        if attempt_count >= 5:
+            print("Max attempts reached. Subprocess may not be ready.")
+    else:
+        print(f"Subprocess '{PROCESS_NAME}' is already running.")
+
+
+def stop_commit_reveal_subprocess():
+    """
+    Stop the commit reveal subprocess if it is running.
+    """
+    pid = get_process(PROCESS_NAME)
+
+    if pid is not None:
+        print(f"Stopping subprocess '{PROCESS_NAME}' with PID {pid}...")
+        os.kill(pid, 15)  # SIGTERM
+        print(f"Subprocess '{PROCESS_NAME}' stopped.")
+    else:
+        print(f"Subprocess '{PROCESS_NAME}' is not running.")
+
+
+class DB:
+    """
+    For ease of interaction with the SQLite database used for --reuse-last and --html outputs of tables
+    """
+
+    def __init__(
+        self,
+        db_path: str = os.path.expanduser("~/.bittensor/bittensor.db"),
+        row_factory=None,
+    ):
+        if not os.path.exists(os.path.dirname(db_path)):
+            os.makedirs(os.path.dirname(db_path))
+
+        self.db_path = db_path
+        self.conn: Optional[sqlite3.Connection] = None
+        self.row_factory = row_factory
+
+    def __enter__(self) -> tuple[sqlite3.Connection, sqlite3.Cursor]:
+        self.conn = sqlite3.connect(self.db_path)
+        self.conn.row_factory = self.row_factory
+        return self.conn, self.conn.cursor()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.conn:
+            self.conn.close()
+
+
+def create_table(title: str, columns: list[tuple[str, str]], rows: list[list]):
+    """
+    Creates and populates the rows of a table in the SQLite database.
+
+    Args:
+        title (str): title of the table.
+        columns (list[tuple[str, str]]): List of tuples where each tuple contains column name and column type.
+        rows (list[list]): List of lists where each sublist contains elements representing a row.
+    """
+    blob_cols = []
+    for idx, (_, col_type) in enumerate(columns):
+        if col_type == "BLOB":
+            blob_cols.append(idx)
+    if blob_cols:
+        for row in rows:
+            for idx in blob_cols:
+                row[idx] = row[idx].to_bytes(row[idx].bit_length() + 7, byteorder="big")
+    with DB() as (conn, cursor):
+        drop_query = f"DROP TABLE IF EXISTS {title}"
+        cursor.execute(drop_query)
+        conn.commit()
+        columns_ = ", ".join([" ".join(x) for x in columns])
+        creation_query = f"CREATE TABLE IF NOT EXISTS {title} ({columns_})"
+        conn.commit()
+        cursor.execute(creation_query)
+        conn.commit()
+        query = f"INSERT INTO {title} ({', '.join([x[0] for x in columns])}) VALUES ({', '.join(['?'] * len(columns))})"
+        cursor.executemany(query, rows)
+        conn.commit()
+    return
```

Comment:
nitpick: I'd return `True` in successful case of table creation. Also, you can use `assert result_of_creation is True` in your code to be make sure operation is successful.